### PR TITLE
MLIBZ-1432: mock unit tests

### PR DESCRIPTION
**Pull request by heyzooi**

### CODE REVIEW

File: Kinvey/Kinvey/Client.swift [hunk 1]
```diff
@@ -72,7 +72,15 @@ open class Client: NSObject, NSCoding, Credential {
         }
     }
     
-    internal var urlSession = URLSession(configuration: URLSessionConfiguration.default) {
+    internal static let urlSessionConfiguration: URLSessionConfiguration = {
```
Comment (tejasranade) [inline Kinvey/Kinvey/Client.swift]: @heyzooi any particular reason to not just use `.default` here?

File: Kinvey/Kinvey/FileStore.swift [hunk 2]
```diff
@@ -219,7 +219,13 @@ open class FileStore {
                 }
                 
                 let handle: (Data?, URLResponse?, Swift.Error?) -> Void = { data, response, error in
-                    if let response = response as? HTTPURLResponse , 200 <= response.statusCode && response.statusCode < 300 {
+                    if self.client.logNetworkEnabled, let response = response as? HTTPURLResponse {
```
Comment (tejasranade) [inline Kinvey/Kinvey/FileStore.swift]: @heyzooi Shouldn't this `logNetworkEnabled` check be a level lower i.e. at the request level?

File: Kinvey/KinveyApp/MICAuthorizationGrantViewController.swift [hunk 3]
```diff
@@ -19,10 +19,14 @@ class MICAuthorizationGrantViewController: UIViewController {
         super.viewDidLoad()
         // Do any additional setup after loading the view, typically from a nib.
         
-        Kinvey.sharedClient.initialize(
-            appKey: "kid_rJVLE1Z5",
-            appSecret: "cd385840cbd94e2caaa8f824c2ff7f46"
-        )
+        if let appKey = ProcessInfo.processInfo.environment["KINVEY_MIC_APP_KEY"],
```
Comment (tejasranade) [inline Kinvey/KinveyApp/MICAuthorizationGrantViewController.swift]: This is great! One suggestion - can you add instructions to the README for someone who is testing this locally vs on Travis?

Comment (heyzooi) [inline Kinvey/KinveyApp/MICAuthorizationGrantViewController.swift]: sure! Doing it now!

File: Kinvey/Kinvey/Client.swift (hunk 1 above)
Comment (heyzooi) [inline Kinvey/Kinvey/Client.swift]: I just reverted this, I think I changed to test something and forgot to revert. Thanks for notice this.

File: Kinvey/Kinvey/FileStore.swift (hunk 2 above)
Comment (heyzooi) [inline Kinvey/Kinvey/FileStore.swift]: You're right, do you prefer to do this change in this PR or create a new ticket to change this in a separate ticket?

Comment (tejasranade) [inline Kinvey/Kinvey/FileStore.swift]: It's up to you. If there's significant work, let's do it separately. I think this PR can be merged either way.

**Pull request merged**